framework/src/multiapps/MultiApp.C

```diff
@@ -1121,22 +1122,30 @@ rankConfig(dof_id_type rank,
 
   if (slot_for_rank[rank] < 0)
     // ranks assigned a negative slot don't have any apps running on them.
-    return {0, 0, 0, 0, false};
-  dof_id_type slot_num = slot_for_rank[rank];
+    return {0, 0, 0, 0, false, 0};
+  processor_id_type slot_num = slot_for_rank[rank];
 
   bool is_first_local_rank = rank == 0 || (slot_for_rank[rank - 1] != slot_for_rank[rank]);
-  auto n_local_apps = apps_per_slot + 1 * (slot_num < leftover_apps);
+  dof_id_type n_local_apps = apps_per_slot + 1 * (slot_num < leftover_apps);
```

Comment:
```suggestion
  processor_id_type n_local_apps = apps_per_slot + 1 * (slot_num < leftover_apps);
```

Reply:
I'd like to keep this, see the reasoning above and let me know if it is acceptable.

Reply:
It is!